strawberryfields/backends/base.py

```diff
@@ -370,6 +364,8 @@ def prepare_squeezed_state(self, r, phi, mode):
     def prepare_displaced_squeezed_state(self, alpha, r, phi, mode):
         r"""Prepare a displaced squeezed state in the specified mode.
 
+        FIXME this method is not currently used by the engine
+
```

Comment:
That's strange :thinking:

Reply:
The `DisplacedSqueezed` Operation uses `backend.prepare_squeezed_state` and then `backend.displacement`, not this method directly.

Reply:
I think it should do what every other Operation does (at least with a local execution), i.e., try to call directly the appropriate backend method, then decompose as needed upon failure

Reply:
:+1:

So, something like the following?

```python
    def _apply(self, reg, backend, **kwargs):
        p = _unwrap(self.p)
        # prepare the displaced squeezed state directly
        backend.prepare_displaced_squeezed_state(p[0], p[1], p[2], *reg)

    def _decompose(self, reg):
        # decompoe into a squeezed state and a displacement
        return [
            Command(Squeezed(self.p[1], self.p[2]), reg),
            Command(Dgate(self.p[0]), reg)
        ]
```

Reply:
Fixed as suggested.